src/deepmol/parameter_optimization/deepchem_hyperparameter_optimization.py

```diff
@@ -66,35 +63,37 @@ def __init__(self,
         self.best_estimator_ = None
         self.cv_results_ = None
 
-    def fit(self, dataset: Dataset):
+    def fit(self, train_dataset: Dataset, validation_dataset: Dataset = None):
         """
         Run hyperparameter search with cross-validation.
 
         Parameters
         ----------
-        dataset: Dataset
+        train_dataset: Dataset
             The dataset to use for the hyperparameter search.
+        validation_dataset: Dataset
+            The dataset to use for validation. If None, use the training dataset.
         """
-        if self.mode != dataset.mode:
+        if self.mode != train_dataset.mode:
             raise ValueError(f'The mode of the model and the dataset must be the same. Got {self.mode} and '
-                             f'{dataset.mode} respectively.')
+                             f'{train_dataset.mode} respectively.')
         results_dict = defaultdict(list)
 
         # split dataset into folds
-        if dataset.mode == 'classification':
+        if train_dataset.mode == 'classification':
```

Comment:
maybe later we can add support for other splitters and for multitask.

Reply:
I will put it in an issue